dv/uvm/icache/doc/ibex_icache_dv_plan.md

```diff
@@ -1,99 +1,142 @@
 
 ### Common DV utility components
-The following utilities provide generic helper tasks and functions to perform activities that are common across the project:
-* [dv_utils_pkg]({{< relref "vendor/lowrisc_ip/dv_utils/README.md" >}})
+
+TBD
 
 ### Compile-time configurations
-[list compile time configurations, if any and what are they used for]
+
+TBD
 
 ### Global types & methods
 All common types and methods defined at the package level can be found in
 `ibex_icache_env_pkg`. Some of them in use are:
-```systemverilog
-[list a few parameters, types & methods; no need to mention all]
-```
-
-###  IBEX_ICACHE Agent
-[Describe here or add link to its README]
-###  IBEX_MEM_INTF_SLAVE Agent
-[Describe here or add link to its README]
-
-### UVC/agent 1
-[Describe here or add link to its README]
-
-### UVC/agent 2
-[Describe here or add link to its README]
 
-
-### Reference models
-[Describe reference models in use if applicable, example: SHA256/HMAC]
+TBD
 
 ### Stimulus strategy
 #### Test sequences
-All test sequences reside in `dv/uvm/icache/dv/env/seq_lib`.
-The `ibex_icache_base_vseq` virtual sequence is extended from `cip_base_vseq` and serves as a starting point.
-All test sequences are extended from `ibex_icache_base_vseq`.
-It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
-Some of the most commonly used tasks / functions are as follows:
-* task 1:
-* task 2:
+
+All test sequences reside in [`dv/uvm/icache/dv/env/seq_lib`](https://github.com/lowRISC/ibex/tree/master/dv/uvm/icache/dv/env/seq_lib).
+
+TBD
 
 #### Functional coverage
-To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
-The following covergroups have been developed to prove that the test intent has been adequately met:
-* cg1:
-* cg2:
+
+TBD
 
 ### Self-checking strategy
 #### Scoreboard
-The `ibex_icache_scoreboard` is primarily used for end to end checking.
-It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
-<!-- explain inputs monitored, flow of data and outputs checked -->
+The `ibex_icache_scoreboard` is used to check that instructions fetched from the cache lie at the correct addresses and contain the correct data.
+It has two analysis ports: one for the memory agent and another for the core agent.
+
+The scoreboard can keep track of what addresses it expects from responses to the core by modelling the cache's internal address counter and following branches.
+
+To check correctness of the fetched data, the scoreboard will keep track of the set of seeds that have been set by the memory agent since the last cache invalidation began.
+When it sees an instruction being passed back to the core, it can check that the response matches one of the possible seeds.
```

Comment:
Is it possible to also compare the data as well as the seed?
From what it sounds like, the seed acts as a timestamp of sorts to check that the cache is pulling data from the correct "set" of valid data, but I think it might also be good to add an extra layer of checking by comparing the data itself as well.

Reply:
Yes, that's the idea.

In the simple case, let's suppose that we haven't yet seen any seed changes, and I know that the memory seed should have been `h'1234`. Then the scoreboard looks at the retrieved address and data. It knows what the address should have been (and checks it's right). Assuming that it is, it can then compute the memory hash function (using some shared code) with seed `h'1234` and the given address and check it equals the data we saw fetched.

I'll update this text to make it a bit clearer what's going on.

Reply:
Sure, that checking approach seems like a solid plan to me.